android/app/src/main/java/io/peggyjo/peggymeter/database/MoodAdapter.java

```diff
@@ -55,6 +55,12 @@ public void onDataChange(DataSnapshot dataSnapshot) {
                         ((Long) entry.get("moodLevel")).intValue(),
                         "" +entry.get("comment")));
             }
+            //set a reference to each entry's id
+            for (int i = 0; i < logs.size(); i++) {
+                logs.get(i).setEntryId(data.keySet().toArray()[i].toString());
```

Comment:
This can be done in one loop.
See https://stackoverflow.com/questions/46898/how-to-efficiently-iterate-over-each-entry-in-a-map

Reply:
I'll check it out.